web/search.dart

```diff
@@ -427,6 +544,17 @@ void initializeSearch(
 
     event.preventDefault();
   });
+
+  // Verifying the href to check if the search html was called to generate the main content elements that are going to be displayed.
+  if (window.location.href.contains('search.html')) {
+    var input = uri.queryParameters['query'];
+    input = htmlEscape.convert(input!);
```

Comment:
If the user just typed this page into the browser, then I think `input!` here will crash in JavaScript, crashing the whole web app. Better to just return early if `input` is null.